tests/lib/file-finder.js

```diff
@@ -160,7 +160,7 @@ describe("FileFinder", () => {
             it("should only find one package.json from the root", () => {
                 expected = path.join(process.cwd(), "package.json");
                 finder = new FileFinder("package.json", process.cwd());
-                actual = finder.findAllInDirectoryAndParents(fileFinderDir);
+                actual = Array.from(finder.findAllInDirectoryAndParents(fileFinderDir)).slice(-1);
 
                 /**
                  * Filter files outside of current workspace, otherwise test fails,
```

Comment:
@not-an-aardvark good callout. I looked inside tests/lib/config.js (https://github.com/eslint/eslint/blob/master/tests/lib/config.js#L531) and I think this test should already test for that since there is the file (tests/fixtures/config-hierarchy/root-true/parent/.eslintrc). I updated the pr to make tests/fixtures/config-hierarchy/root-true/parent/.eslintrc and bad configuration file instead. If I take out root:true in tests/fixtures/config-hierarchy/root-true/parent/root/.eslintrc, the test will fail with below screen:
<img width="1166" alt="screen shot 2017-05-28 at 11 28 51 pm" src="https://cloud.githubusercontent.com/assets/3211873/26535671/a1bc167a-43fd-11e7-888b-e9f4529058d0.png">